auto_inertial_params/proposal.md

```diff
@@ -0,0 +1,149 @@
+* [User Perspective](#user-perspective): Describes the current usage and the proposed usage by describing the terms to be added in the SDFormat specification
+* [Proposed Implmentation](#proposed-implementation): Detailed explanation of the proposed implmentation and modifications to be done in the C++ API of requried libraries like libsdformat. 
+
+## Motivation
+
+Currently, there are 2 major workflows used by the users to obtain the correct inertial parameters of their models:
+
+ * Using CAD softwares like [Fusion360](https://www.autodesk.in/products/fusion-360/overview?term=1-YEAR&tab=subscription) or [Solidworks](https://www.solidworks.com/). Many users design their robot models using such CAD software which usually provide plugins that automatically generates the URDF/SDF for their model. These plugins handle the calculation of the inertial parameters. For eg, Fusion360 provides the [Fusion2URDF](https://github.com/syuntoku14/fusion2urdf) plugin which automatically generates a URDF with all the inertial parameters.
+
+ * Another way is to use 3rd-party Mesh Processing Software like [Meshlab](https://www.meshlab.net/). Such softwares take the mesh file as an input and provide the inertial parameters as an output which can then be copied and pasted into the URDF/SDF file. This is also the method that was suggested in official [Classic Gazebo docs](https://classic.gazebosim.org/tutorials?cat=build_robot&tut=inertia).
+
+Both of these ways create a dependency on external software and might be complicated for beginners. Native support for this feature directly into Gz Sim would facilitate the effortless generation of accurate simulations.
+
+## User Perspective
+
+To specify the `<inertial>` element of a `<link>` in SDFormat, the user needs to add the `<mass>`, `<pose>`, and `<inertia>` tags. Here `<mass>` is the mass of the link and `<pose>` is the pose of the Centre of Mass with respect to the link frame. The `<inertia>` tag, on the other hand, needs to enclose the following 6 tags:
+
+ * `<ixx>` (Moment of Inertia)
+
+ * `<iyy>` (Moment of Inertia)
+
+ * `<izz>` (Moment of Inertia)
+
+ * `<ixy>` (Product of Inertia)
+ 
+ * `<ixz>` (Product of Inertia)
+ 
+ * `<izy>` (Product of Inertia)
+
+This proposal suggests the addition of the following in `SDFormat`: 
+
+ 1. An `auto` parameter for the `<inertia>` tag that would tell `libsdformat` to calculate Inertia matrix values automatically for the respective link. 
+
+ 2. A `//link/collision/material_density` tag for specifying the density of the collision geometry. This density value would be used to calculate the inertial parameters of the respective collision geometry. Adding this as part of the `<collision>` tag would allow a user to simulate links with different material types for different collisions. By default, the value of density would be set equal to that of water which is 1 kg/m^3.  
```

Comment:
`material_density` seems a little verbose. Why not just `density` or `mass_density`?

Reply:
Changed the tag in propsal in 3490edd2fb5a536c170f15ffb117190ae3f788e9 and in sdformat in c3c52aa223b69e6c406b327ab2c3cd1d8c2cb152